sycl/include/CL/sycl/stream.hpp

```diff
@@ -113,15 +125,37 @@ class stream {
   friend decltype(Obj::impl) detail::getSyclObjImpl(const Obj &SyclObject);
 #endif
 
-  // Accessor to stream buffer
-  mutable detail::stream_impl::AccessorType Acc;
+  // Accessor to the global stream buffer. Global buffer contains all output
+  // from the kernel.
+  mutable detail::stream_impl::GlobalBufAccessorT GlobalBuf;
 
-  // Atomic accessor to the offset variable. It represents an offset in the
-  // stream buffer.
-  mutable detail::stream_impl::OffsetAccessorType OffsetAcc;
-  mutable stream_manipulator Manipulator = defaultfloat;
+  // Atomic accessor to the global offset variable. It represents an offset in
+  // the global stream buffer. Since work items will flush data to global buffer
+  // in parallel we need atomic access to this offset.
+  mutable detail::stream_impl::GlobalOffsetAccessorT GlobalOffset;
+
+  // Accessor to the pool of flush buffers. Flush buffer contains output from
+  // work item in the work group. One flush buffer per work item in the work
+  // group.
+  mutable detail::stream_impl::FlushBufAccessorT FlushBufs;
+
+  // Each work item in the work group writes to its own flush buffer in the
+  // pool. This accessor is used to atomically get offset of the flush buffer in
+  // the pool for each work item in the work group. This approach is used
+  // because currently it is not possible to get work item id in the work group
+  // without id object, which is passed to the kernel.
+  mutable detail::stream_impl::LocalOffsetAccessorT WIOffsetAcc;
+
+  // Offset of the WI's flush buffer in the pool.
+  mutable unsigned WIOffset = 0;
+
+  // Offset in the flush buffer
+  mutable unsigned Offset = 0;
+
+  mutable size_t FlushBufferSize = 200;
```

Comment:
Why `200`?

Reply:
Just used this constant during development for testing purposes, forgot to remove before uploading. Thanks, fixed.